src/Spago/Command/Auth.purs

```diff
@@ -0,0 +1,56 @@
+module Spago.Command.Auth where
+
+import Spago.Prelude
+
+import Data.Array as Array
+import Data.String (Pattern(..))
+import Data.String as String
+import Node.Path as Path
+import Registry.SSH as SSH
+import Spago.Command.Fetch (FetchEnv)
+import Spago.Config as Config
+import Spago.FS as FS
+
+type AuthArgs = { keyPath :: FilePath }
+
+run :: AuthArgs -> Spago (FetchEnv _) Unit
+run { keyPath } = do
+  logDebug $ "Authenticating with key at path " <> keyPath
+  let
+    -- we don't want to accidentally read the private key, so we always point to the public
+    path = case String.stripSuffix (Pattern ".pub") keyPath of
+      Just _rest -> keyPath
+      Nothing -> keyPath <> ".pub"
+
+  newOwner <- FS.exists path >>= case _ of
```

Comment:
When I first read this file I was wondering why this was necessarily a "new owner", until I saw that this function is specifically intended as a command for adding a new key. Could that be added as a module comment so someone coming into this file without a lot of context knows its purpose?